Sources/NostrSDK/Events/Tags/EventCoordinates.swift

```diff
@@ -86,16 +90,21 @@ public struct EventCoordinates: PubkeyProviding, RelayProviding, RelayURLValidat
     }
 
     /// Initializes coordinates to a replaceable event.
+    /// Returns nil if the kind is not a replaceable event kind.
     /// - Parameters:
     ///   - kind: The ``EventKind`` of the referenced replaceable event.
     ///   - pubkey: The pubkey that signed the referenced replaceable event.
-    ///   - identifier: The identifier of the referenced replaceable event.
+    ///   - identifier: The identifier of the referenced replaceable event. Must be `nil` if `kind.isNonParameterizedReplaceable` is `true`. Must not be `nil` if `kind.isParameterizedReplaceable` is `true`.
     ///   - relayURL: A relay in which the referenced replaceable event could be found.
-    public init?(kind: EventKind, pubkey: PublicKey, identifier: String?, relayURL: URL? = nil) {
+    public init?(kind: EventKind, pubkey: PublicKey, identifier: String? = nil, relayURL: URL? = nil) throws {
+        guard (kind.isParameterizedReplaceable && identifier != nil) || (kind.isNonParameterizedReplaceable && identifier == nil) else {
+            throw EventCoordinatesError.invalidInput
+        }
+
         let otherParameters: [String]
         if let relayURL {
             guard (try? RelayURLValidator.shared.validateRelayURL(relayURL)) != nil else {
-                return nil
+                throw EventCoordinatesError.invalidInput
             }
 
             otherParameters = [relayURL.absoluteString]
```

Comment:
It would be better to allow `validateRelayURL` to throw it's own, more specific errors than to throw them away and throw a more generic one here.

```swift
            let validatedURL = try RelayURLValidator.shared.validateRelayURL(relayURL)
            otherParameters = [validatedURL.absoluteString]
```